demos/common/cpp/models/src/image_processing_model.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+// Copyright (C) 2018-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "models/image_processing_model.h"
+#include "utils/ocv_common.hpp"
+
+using namespace InferenceEngine;
+
+ImageProcessingModel::ImageProcessingModel(const std::string& modelFileName, const cv::Size inputImageShape) :
+    ModelBase(modelFileName) {
+        if (inputImageShape == cv::Size(0, 0)) {
+            type = "super_resolution";
+            viewInfo = cv::Size(1920, 1080);
+        } else {
+            type = "deblurring";
+            inputSize = inputImageShape;
+            viewInfo = inputSize;
+        }
```

Comment:
As for me, it is better to implement these with different classes, inherited from `ImageProcessingModel`

Reply:
Created separate classes.